### PR TITLE
replaced mock data

### PR DESCRIPTION
**Pull request by casterweb**

### CODE REVIEW

File: src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.ts [hunk 1]
```diff
@@ -20,16 +21,16 @@ export class HrProfileDialogComponent {
   };
 
   constructor(
-    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    @Inject(MAT_DIALOG_DATA) public data: ApiAssignTest,
     private readonly postHttpService: PostHttpService,
   ) {}
 
-  assignTest() {
+  assignTestUser() {
```
Comment (RomPavel) [inline src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.ts:28]: Can you assign test without choosen deadline?
What will happen?

File: src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.html [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+      [matDatepicker]="picker"
+    />
     <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
     <mat-datepicker #picker></mat-datepicker>
   </mat-form-field>
 
   <div class="dialog-button">
     <button mat-raised-button mat-dialog-close cdkFocusInitial>Close</button>
-    <button mat-raised-button color="primary" (click)="assignTest()">Assign Test</button>
+    <button mat-raised-button mat-dialog-close color="primary" (click)="assignTestUser()">
```
Comment (RomPavel) [inline src/app/models/hr-profile/hr-profile-dialog/hr-profile-dialog.component.html:20]: Check if modal closes after assign

File: src/app/models/hr-profile/hr-profile.component.ts [hunk 3]
```diff
@@ -90,7 +90,9 @@ export class HrProfileComponent implements OnInit {
     });
   }
 
-  onClick() {
-    this.dialog.open(HrProfileDialogComponent);
+  onClick(userId: string) {
+    this.dialog.open(HrProfileDialogComponent, {
+      data: { userId },
```
Comment (Inunor) [inline src/app/models/hr-profile/hr-profile.component.ts:95]: I suggest to add here data: { userId } as ApiAssignTest. As I remember angular material dialog doesn't check data interface, so you can easily misspell variable names and pass wrong data

**Review by Inunor — APPROVED** [review]

File: src/app/models/hr-profile/hr-profile.component.ts (hunk 3 above)
Comment (casterweb) [reply src/app/models/hr-profile/hr-profile.component.ts]: https://github.com/Inunor/internship/pull/157/commits/28c4cb2996931a32a07007cc3329b730ce552552

**Pull request merged**